agent/info.go

```diff
@@ -0,0 +1,322 @@
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-process-agent/config"
+	"github.com/DataDog/datadog-process-agent/model"
+	"github.com/DataDog/datadog-process-agent/util"
+)
+
+var (
+	infoMutex           sync.RWMutex
+	infoOnce            sync.Once
+	infoStart           = time.Now()
+	infoNotRunningTmpl  *template.Template
+	infoTmpl            *template.Template
+	infoErrorTmpl       *template.Template
+	infoDockerSocket    string
+	infoLastCollectTime string
+	infoProcCount       int
+	infoContainerCount  int
+	infoQueueSize       int
+)
+
+const (
+	infoTmplSrc = `{{.Banner}}
+{{.Program}}
+{{.Banner}}
+
+  Pid: {{.Status.Pid}}
+  Hostname: {{.Status.Config.HostName}}
+  Uptime: {{.Status.Uptime}} seconds
+  Mem alloc: {{.Status.MemStats.Alloc}} bytes
+
+  Last collection time: {{.Status.LastCollectTime}}{{if ne .Status.DockerSocket ""}}
+  Docker socket: {{.Status.DockerSocket}}{{end}}
+  Number of processes: {{.Status.ProcessCount}}
+  Number of containers: {{.Status.ContainerCount}}
+  Queue length: {{.Status.QueueSize}}
+
+  Logs: {{.Status.Config.LogFile}}{{if .Status.Config.Proxy}}
+  HttpProxy: {{.Status.Config.Proxy}}{{end}}{{if ne .Status.ContainerId ""}}
+  Container ID: {{.Status.ContainerId}}{{end}}
+
+`
+	infoNotRunningTmplSrc = `{{.Banner}}
+{{.Program}}
+{{.Banner}}
+
+  Not running
+
+`
+	infoErrorTmplSrc = `{{.Banner}}
+{{.Program}}
+{{.Banner}}
+
+  Error: {{.Error}}
+
+`
+)
+
+func publishUptime() interface{} {
+	return int(time.Since(infoStart) / time.Second)
+}
+
+func publishVersion() interface{} {
+	return infoVersion{
+		Version:   Version,
+		GitCommit: GitCommit,
+		BuildDate: BuildDate,
+		GoVersion: GoVersion,
+	}
+}
+
+func publishDockerSocket() interface{} {
+	infoMutex.RLock()
+	defer infoMutex.RUnlock()
+	return infoDockerSocket
+}
+
+func updateDockerSocket(path string) {
+	infoMutex.Lock()
+	defer infoMutex.Unlock()
+	infoDockerSocket = path
+}
+
+func publishLastCollectTime() interface{} {
+	infoMutex.RLock()
+	defer infoMutex.RUnlock()
+	return infoLastCollectTime
+}
+
+func updateLastCollectTime(t time.Time) {
+	infoMutex.Lock()
+	defer infoMutex.Unlock()
+	infoLastCollectTime = t.Format("2006-01-02 15:04:05")
+}
+
+func publishProcCount() interface{} {
+	infoMutex.RLock()
+	defer infoMutex.RUnlock()
+	return infoProcCount
+}
+
+func publishContainerCount() interface{} {
+	infoMutex.RLock()
+	defer infoMutex.RUnlock()
+	return infoContainerCount
+}
+
+func updateProcContainerCount(msgs []model.MessageBody) {
+	var count int
+	// this is the flag to determine what type of collections are we dealing with
+	var isProc bool
+	for _, m := range msgs {
+		switch msg := m.(type) {
+		case *model.CollectorContainerRealTime:
+			isProc = false
+			count += len(msg.Stats)
+		case *model.CollectorContainer:
+			isProc = false
+			count += len(msg.Containers)
+		case *model.CollectorRealTime:
+			isProc = true
+			count += len(msg.Stats)
+		case *model.CollectorProc:
```

Comment:
CollectorProc has both containers and processes

Reply:
What about container only situation?

Reply:
Only `CollectorContainer`